Libraries/LLM/Load.swift

```diff
@@ -11,82 +11,91 @@ struct LLMError: Error {
-        let enumerator = FileManager.default.enumerator(
-            at: modelDirectory, includingPropertiesForKeys: nil)!
-        for case let url as URL in enumerator {
-            if url.pathExtension == "safetensors" {
-                let w = try loadArrays(url: url)
-                for (key, value) in w {
-                    weights[key] = value
-                }
+func loadSynchronous(modelDirectory: URL) throws -> LLMModel {
+    // create the model (no weights loaded)
+    let configurationURL = modelDirectory.appending(component: "config.json")
+    let baseConfig = try JSONDecoder().decode(
+        BaseConfiguration.self, from: Data(contentsOf: configurationURL))
+
+    let model = try baseConfig.modelType.createModel(configuration: configurationURL)
+
+    // load the weights
+    var weights = [String: MLXArray]()
+    let enumerator = FileManager.default.enumerator(
+        at: modelDirectory, includingPropertiesForKeys: nil)!
+    for case let url as URL in enumerator {
+        if url.pathExtension == "safetensors" {
+            let w = try loadArrays(url: url)
+            for (key, value) in w {
+                weights[key] = value
             }
         }
+    }
 
-        // per-model cleanup
-        weights = model.sanitize(weights: weights)
+    // per-model cleanup
+    weights = model.sanitize(weights: weights)
 
-        // quantize if needed
-        if let quantization = baseConfig.quantization {
-            quantize(model: model, groupSize: quantization.groupSize, bits: quantization.bits) {
-                path, module in
-                weights["\(path).scales"] != nil
-            }
+    // quantize if needed
+    if let quantization = baseConfig.quantization {
+        quantize(model: model, groupSize: quantization.groupSize, bits: quantization.bits) {
+            path, module in
+            weights["\(path).scales"] != nil
         }
+    }
 
-        // apply the loaded weights
-        let parameters = ModuleParameters.unflattened(weights)
-        try model.update(parameters: parameters, verify: [.all])
-
-        eval(model)
+    // apply the loaded weights
+    let parameters = ModuleParameters.unflattened(weights)
+    try model.update(parameters: parameters, verify: [.all])
 
-        return (model, tokenizer)
+    eval(model)
 
-    } catch Hub.HubClientError.authorizationRequired {
-        // an authorizationRequired means (typically) that the named repo doesn't exist on
-        // on the server so retry with local only configuration
-        var newConfiguration = configuration
-        newConfiguration.id = .directory(configuration.modelDirectory(hub: hub))
-        return try await load(
-            hub: hub, configuration: newConfiguration, progressHandler: progressHandler)
-    }
+    return model
 }
 
+/// Load and return the model and tokenizer wrapped in a ``ModelContainer`` (provides
+/// thread safe access).
 public func loadModelContainer(
```

Comment:
Same factory method as before -- this loads a model and produces a `ModelContainer` (actor) that allows thread safe access to the model/tokenizer.

The interior has been redone to avoid passing non-Sendable data between threads/isolation contexts.